.github/test-pre-script.sh

```diff
@@ -3,4 +3,15 @@
 #  See LICENSE file for licensing details.
 
 # Call script with sudo to install the required native libraries for installing the plugin's dependencies
-sudo bash -xe "$(dirname "$0")"/../install-libs.sh
\ No newline at end of file
+sudo bash -xe "$(dirname "$0")"/../install-libs.sh
+
+# Install python 3.12.2
+# We don't activate after installing since this is not required by default
+# python 3.12.3 will be installed in $HOME/.pyenv/versions/3.12.2/bin
+curl https://pyenv.run | bash
+export PYENV_ROOT="$HOME/.pyenv"
+[[ -d $PYENV_ROOT/bin ]] && export PATH="$PYENV_ROOT/bin:$PATH"
+eval "$(pyenv init -)"
+eval "$(pyenv virtualenv-init -)"
+pyenv install 3.12.2
+pyenv doctor
```

Comment:
It would probably be a good idea for a follow-up idea if we can use `actions/setup-python` in `operator-workflows` to manage Python versions

Reply:
good point, I wasn't aware of this. Will take a look if there are other repos that need multiple python versions

Reply:
Yes, sometimes for charm libraries, I want to run unit tests across different Python versions to ensure compatibility with all LTS versions. It would be very helpful if there were a convenient way to set the Python version.